### PR TITLE
petclinic tests

### PR DESCRIPTION
**Pull request by hurelhuyag**
- [x] Entity tests
- [x] Projection tests
- [x] Query tests
- [x] Resource handler tests

### CODE REVIEW

File: vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/SpecialtyTypeEntityTests.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        registry.register(new SourcedTypeRegistry.Info(journal, SpecialtyTypeEntity.class, SpecialtyTypeEntity.class.getSimpleName()));
+
+        specialtyType = world.actorFor(SpecialtyType.class, SpecialtyTypeEntity.class, "#1");
+    }
+
+    @Test
+    public void offer(){
+        final AccessSafely dispatcherAccess = dispatcher.afterCompleting(1);
+
+        SpecialtyTypeState petState = specialtyType.offer("Behaviour").await();
```
Comment (VaughnVernon) [inline vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/SpecialtyTypeEntityTests.java:59]: Fixture needed.

File: vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/SpecialtyTypeEntityTests.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        // this will block until the first event is persisted in the Journal
+        assertEquals(1, (int) dispatcherAccess.readFrom("entriesCount"));
+        BaseEntry<String> appendedAt0 = dispatcherAccess.readFrom("appendedAt", 0);
+        assertNotNull(appendedAt0);
+        assertEquals(SpecialtyTypeOffered.class.getName(), appendedAt0.typeName());
+    }
+
+    @Test
+    public void rename(){
+        offer();
```
Comment (VaughnVernon) [inline vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/SpecialtyTypeEntityTests.java:73]: Fixture needed.

File: vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/VeterinarianEntityTests.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        registry.register(new SourcedTypeRegistry.Info(journal, VeterinarianEntity.class, VeterinarianEntity.class.getSimpleName()));
+
+        vet = world.actorFor(Veterinarian.class, VeterinarianEntity.class, "#1");
+    }
+
+    @Test
+    public void offer(){
+        final AccessSafely dispatcherAccess = dispatcher.afterCompleting(1);
+
+        final ContactInformation contact = ContactInformation.of(
```
Comment (VaughnVernon) [inline vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/VeterinarianEntityTests.java:64]: Fixture needed.

File: vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/VeterinarianEntityTests.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        // this will block until the first event is persisted in the Journal
+        assertEquals(1, (int) dispatcherAccess.readFrom("entriesCount"));
+        BaseEntry<String> appendedAt0 = dispatcherAccess.readFrom("appendedAt", 0);
+        assertNotNull(appendedAt0);
+        assertEquals(VeterinarianRegistered.class.getName(), appendedAt0.typeName());
+    }
+
+    @Test
+    public void rename(){
+        offer();
```
Comment (VaughnVernon) [inline vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/VeterinarianEntityTests.java:97]: Fixture needed.

File: vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/VeterinarianEntityTests.java [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        // this will block until the first event is persisted in the Journal
+        assertEquals(2, (int) dispatcherAccess.readFrom("entriesCount"));
+        BaseEntry<String> appendedAt1 = dispatcherAccess.readFrom("appendedAt", 1);
+        assertNotNull(appendedAt1);
+        assertEquals(VeterinarianSpecialtyChosen.class.getName(), appendedAt1.typeName());
+    }
+
+    @Test
+    public void contactUpdate(){
+        offer();
```
Comment (VaughnVernon) [inline vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/VeterinarianEntityTests.java:134]: Fixture needed.

File: vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/AnimalTypeEntityTests.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+import io.vlingo.symbio.store.journal.inmemory.InMemoryJournalActor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class AnimalTypeEntityTests {
```
Comment (VaughnVernon) [inline vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/AnimalTypeEntityTests.java:21]: It appears that non of your tests test each of the factory methods in the protocol interfaces. Please add simple tests for each.

**Review by VaughnVernon — CHANGES_REQUESTED** [review]
@hurelhuyag There are a number of required changes, but most of them are repeats of others.

File: vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/AnimalTypeEntityTests.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    private Journal<String> journal;
+    private MockJournalDispatcher dispatcher;
+    private SourcedTypeRegistry registry;
+
+    private AnimalType pet;
+
+    @BeforeEach
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public void setUp(){
+        testWorld = TestWorld.startWithDefaults("test-es");
```
[the remark this answers is not in the document]
Comment (hurelhuyag) [reply vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/AnimalTypeEntityTests.java]: 51d4fd1e

**Comment by hurelhuyag** [comment]
FIxtures ef82a89c

File: vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/AnimalTypeEntityTests.java (hunk 6 above)
Comment (hurelhuyag) [reply vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/AnimalTypeEntityTests.java:20]: What do you mean simple test for protocol interface? Isn't this protocol method calling?

https://github.com/hurelhuyag/vlingo-examples/blob/master/vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/AnimalTypeEntityTests.java#L72

Comment (VaughnVernon) [reply vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/AnimalTypeEntityTests.java:20]: I meant the interface static factory methods.

Comment (hurelhuyag) [reply vlingo-petclinic/src/test/java/io/vlingo/developers/petclinic/AnimalTypeEntityTests.java:20]: fixed by b7b25b44

**Pull request merged**